doc/community/developer-guide.md

```diff
@@ -76,59 +106,120 @@ To implement such behavior, you can use any of the functions defined in `sql.exc
+These errors that hide the traceback should only be used in the `%sql`/`%%sql` magic context. For example, in our ggplot API (Python-based), we do not hide tracebacks as users might need them to debug their code
 ```
 
-+++ {"user_expressions": []}
++++
 
-### Unit testing custom errors
+## Getting connections
 
-The internal implementation of `sql.exceptions` is a workaround due to some IPython limitations; in consequence, you need to test for `IPython.error.UsageError` when testing, see `test_util.py` for examples, and `exceptions.py` for more details.
+When adding features to JupySQL magics (`%sql/%%sql`), you can use the `ConnectionManager` to get the current open connections.
 
-+++
+```{code-cell} ipython3
+:tags: [remove-output]
 
-### Handling common errors
+%load_ext sql
+```
 
-Currently, these common errors are handled by providing more meaningful error messages:
+```{code-cell} ipython3
+import sqlite3
 
-* Syntax error in SQL query - The SQL query is parsed using `sqlglot` and possible syntax issues or query suggestions are provided to the user. This raises a `UsageError` with the message.
-* Missing password when connecting to PostgreSQL - Displays guide on DB connections
-* Invalid connection strings when connecting to DuckDB.
+conn = sqlite3.connect("")
 
-+++
+%sql sqlite:// --alias sqlite-sqlalchemy
+%sql conn --alias sqlite-dbapi
+```
 
-## Managing Connections
+We can access the current connection using `ConnectionManager.current`:
 
-In our codebase, we establish connections to databases with `SQLAlchemyConnection` and `DBAPIConnection` objects (they have the same interface).
+```{code-cell} ipython3
+from sql.connection import ConnectionManager
 
-Furthermore, we have a `ConnectionManager` to manage all the connections.
+conn = ConnectionManager.current
+conn
+```
+
+To get all open connections:
 
-### Working with connections
+```{code-cell} ipython3
+ConnectionManager.connections
+```
 
-We can access the current connection using `ConnectionManager.current`.
+## Using connections
+
+Connections are either `SQLAlchemyConnection` or `DBAPIConnection` object. Both have the same interface, the difference is that the first one is a connectioon established via SQLAlchemy and `DBAPIConnection` one is a connection established by an object that follows the [Python DB API](https://peps.python.org/pep-0249/).
 
 ```{code-cell} ipython3
-:tags: [remove-output]
+conn_sqlalchemy = ConnectionManager.connections["sqlite-sqlalchemy"]
+conn_dbapi = ConnectionManager.connections["sqlite-dbapi"]
+```
 
-%load_ext sql
-%sql sqlite://
+### `raw_execute`
+
+```{important}
+Always use `raw_execute` for user-submitted queries!
 ```
 
+`raw_execute` allows you to execute a given SQL query in the connection. Unlike `execute`, `raw_execute` does not perform any transpilation.
```

Comment:
Might help to add a brief description about what `transpilation` does

Reply:
added a link to the last section, which explains transpilation